File: src/main/java/fi/quanfoxes/assembler/Builder.java
package fi.quanfoxes.assembler;

public class Builder {
    private StringBuilder builder = new StringBuilder();

    public Builder() {}
    
    public Builder(String text) {
        append(text);
    }

    public Builder comment(String comment) {
        builder = builder.append("; ").append(comment).append("\n");
        return this;
    }

    public Builder comment(String format, Object... args) {
        builder = builder.append("; ").append(String.format(format, args)).append("\n");
        return this;
    }

    public Builder append(String text) {
        builder = builder.append(text).append("\n");
        return this;
    }

    public Builder append(String format, Object... args) {
        builder = builder.append(String.format(format, args)).append("\n");
        return this;
    }

    @Override
    public String toString() {
        return builder.toString() + "\n";
    }
}

File: Zigzag/Assembler/Builder.cs
using System.Text;

public class Builder
{
	private StringBuilder Buffer = new StringBuilder();

	public Builder(string text = "")
	{
		if (text.Length > 0)
		{
			Append(text);
		}
	}

	public Builder Comment(string comment)
	{
		Buffer = Buffer.Append("; ").Append(comment).Append("\n");
		return this;
	}

	public Builder Comment(string format, params object[] args)
	{
		Buffer = Buffer.Append("; ").Append(string.Format(format, args)).Append("\n");
		return this;
	}

	public Builder Append(Builder builder)
	{
		Buffer = Buffer.Append(builder.ToString()).Append("\n");
		return this;
	}

	public Builder Append(string text)
	{
		Buffer = Buffer.Append(text).Append("\n");
		return this;
	}

	public Builder Append(string format, params object[] args)
	{
		Buffer = Buffer.Append(string.Format(format, args)).Append("\n");
		return this;
	}

	public void Break()
	{
		Buffer = Buffer.Append("\n\n");
	}

	public override string ToString()
	{
		return Buffer.ToString() + "\n";
	}
}